src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -130,6 +130,35 @@ const updateOrInsertTx = (txHistory: TxHistory, tx: Tx, accountId: AccountId) =>
   )
 }
 
+const updateOrInsertTxs = (txHistory: TxHistory, incomingTxs: Tx[], accountId: AccountId) => {
+  const { txs } = txHistory
+
+  const filteredIncomingTxsWithIndex = incomingTxs
+    .filter(tx => !checkIsSpam(tx))
+    .map(tx => {
+      const txIndex = serializeTxIndex(accountId, tx.txid, tx.pubkey, tx.data)
+      return { tx, txIndex }
+    })
+
+  for (const { txIndex, tx } of filteredIncomingTxsWithIndex) {
+    const isNew = !txs.byId[txIndex]
+
+    if (!isNew) continue
+
+    // update or insert tx
+    txs.byId[txIndex] = tx
+
+    // for a given tx, find all the related assetIds, and keep an index of
+    // txids related to each asset id
+    getRelatedAssetIds(tx).forEach(relatedAssetId =>
+      deepUpsertArray(txs.byAccountIdAssetId, accountId, relatedAssetId, txIndex),
+    )
+  }
+
+  const getBlockTime = ([_, tx]: [string, Tx]) => tx.blockTime
+  txs.ids = orderBy(Object.entries(txs.byId), getBlockTime, 'desc').map(([txIndex, _]) => txIndex)
+}
```

Comment:
Since we're upserting more frequently, this change is intended to minimise the compute impact of doing so